src/compose/service-manager.ts

```diff
@@ -32,626 +32,616 @@ type ServiceManagerEventEmitter = StrictEventEmitter<
+	);
+	await kill(current);
+}
+
+export async function killAllLegacy(): Promise<void> {
+	// Containers haven't been normalized (this is an updated supervisor)
+	const supervisorImageId = (
+		await docker.getImage(constants.supervisorImage).inspect()
+	).Id;
+
+	for (const container of await docker.listContainers({ all: true })) {
+		if (container.ImageID !== supervisorImageId) {
+			await killContainer(container.Id, {
+				serviceName: 'legacy',
+			});
+		}
 	}
+}
 
-	public async handover(current: Service, target: Service) {
-		// We set the running container to not restart so that in case of a poweroff
-		// it doesn't come back after boot.
-		await this.prepareForHandover(current);
-		await this.start(target);
-		await this.waitToKill(
-			current,
-			target.config.labels['io.balena.update.handover-timeout'],
+export function kill(service: Service, opts: KillOpts = {}) {
+	if (service.containerId == null) {
+		throw new InternalInconsistencyError(
+			`Attempt to kill container without containerId! Service :${service}`,
 		);
-		await this.kill(current);
 	}
+	return killContainer(service.containerId, service, opts);
+}
 
-	public async killAllLegacy(): Promise<void> {
-		// Containers haven't been normalized (this is an updated supervisor)
-		// so we need to stop and remove them
-		const supervisorImageId = (
-			await docker.getImage(constants.supervisorImage).inspect()
-		).Id;
-
-		for (const container of await docker.listContainers({ all: true })) {
-			if (container.ImageID !== supervisorImageId) {
-				await this.killContainer(container.Id, {
-					serviceName: 'legacy',
-				});
-			}
-		}
+export async function remove(service: Service) {
+	logger.logSystemEvent(LogTypes.removeDeadService, { service });
+	const existingService = await get(service);
+
+	if (existingService.containerId == null) {
+		throw new InternalInconsistencyError(
+			`No containerId provided for service ${service.serviceName} in ServiceManager.updateMetadata. Service: ${service}`,
+		);
 	}
 
-	public kill(service: Service, opts: KillOpts = {}) {
-		if (service.containerId == null) {
-			throw new InternalInconsistencyError(
-				`Attempt to kill container without containerId! Service :${service}`,
-			);
+	try {
+		await docker.getContainer(existingService.containerId).remove({ v: true });
+	} catch (e) {
+		if (!NotFoundError(e)) {
+			logger.logSystemEvent(LogTypes.removeDeadServiceError, {
+				service,
+				error: e,
+			});
+			throw e;
 		}
-		return this.killContainer(service.containerId, service, opts);
 	}
+}
+export function getAllByAppId(appId: number) {
+	return getAll(`app-id=${appId}`);
+}
 
-	public async remove(service: Service) {
-		logger.logSystemEvent(LogTypes.removeDeadService, { service });
-		const existingService = await this.get(service);
+export async function stopAllByAppId(appId: number) {
+	for (const app of await getAllByAppId(appId)) {
+		await kill(app, { removeContainer: false });
```

Comment:
Is this intended to be done in serial?

Reply:
bump

Reply:
yeah sorry I hadn't submitted my comments.

Reply:
Right, but is it intended to be done in serial or is it a mistake that passed through previous code review?